app/src/main/java/com/noalecohen/dispatcher/viewmodel/ArticlesViewModel.kt

```diff
@@ -1,46 +1,61 @@
 
     val articlesLiveData: MutableLiveData<List<Article>> = MutableLiveData()
-    val requestStateLiveData: MutableLiveData<RequestState> = MutableLiveData(RequestState.Idle)
+    val searchArticlesLiveData: MutableLiveData<List<Article>> = MutableLiveData()
+    val articlesStateLiveData: MutableLiveData<RequestState> = MutableLiveData(RequestState.Idle)
+    val searchArticlesStateLiveData: MutableLiveData<RequestState> =
+        MutableLiveData(RequestState.Idle)
+
+    init {
+        fetchTopHeadlinesByCountry(DEFAULT_COUNTRY_CODE)
+    }
 
     fun fetchTopHeadlinesByCountry(country: String) {
-        articlesRepository.fetchTopHeadlinesByCountry(country).enqueue(object : Callback<News> {
-            override fun onResponse(call: Call<News>, response: Response<News>) {
-                if (response.isSuccessful) {
-                    val apiResponse = response.body()
-                    articlesLiveData.postValue(apiResponse?.articles)
-                    requestStateLiveData.postValue(RequestState.Success)
+
+        articlesRepository.fetchTopHeadlinesByCountry(country) { result, error ->
+            if (result.isNotEmpty()) {
+                articlesLiveData.postValue(result)
+                articlesStateLiveData.postValue(RequestState.Success)
+
+            } else {
+                if (error == null) {
+                    articlesStateLiveData.postValue(RequestState.Success)
                 } else {
-                    val errorResponse = Gson().fromJson(
-                        response.errorBody()?.charStream(),
-                        ErrorResponse::class.java
-                    )
-                    requestStateLiveData.postValue(RequestState.Error(errorResponse.message))
-                    Log.d("Request Error", errorResponse.message)
+                    articlesStateLiveData.postValue(error.let { RequestState.Error(it) })
                 }
             }
+        }
+    }
 
-            override fun onFailure(call: Call<News>, t: Throwable) {
-                requestStateLiveData.postValue(RequestState.Error(t.toString()))
-                Log.d("Network Error", t.localizedMessage)
+    fun fetchFilterResults(keyword: String) {
+
+        articlesRepository.fetchFilterResults(keyword) { result, error ->
+            if (result.isNotEmpty()) {
+                searchArticlesLiveData.postValue(result)
+                searchArticlesStateLiveData.postValue(RequestState.Success)
+            } else {
+                if (error == null) {
+                    searchArticlesLiveData.postValue(emptyList())
+                    searchArticlesStateLiveData.postValue(RequestState.Success)
+                } else {
+                    searchArticlesLiveData.postValue(emptyList())
+                    searchArticlesStateLiveData.postValue(error.let { RequestState.Error(it) })
+                }
             }
-        })
+        }
+
     }
 
+    companion object {
+        const val DEFAULT_COUNTRY_CODE = "us"
```

Comment:
this is ok but a better version will be to take the device country code.
Something like this:
`
context.getResources().getConfiguration().locale.getCountry();
`